packages/react/src/components/tag/Tag.tsx

```diff
@@ -6,138 +6,177 @@ import { IconCross } from '../../icons';
 import { useTheme } from '../../hooks/useTheme';
 
-export interface TagCustomTheme {
-  '--tag-background'?: string;
-  '--tag-color'?: string;
-  '--tag-focus-outline-color'?: string;
+export interface TagTheme {
+  '--background-color-hover'?: string;
+  '--background-color'?: string;
+  '--border-color-action'?: string;
+  '--color'?: string;
+  '--outline-color-focus'?: string;
+}
+
+export enum TagVariant {
+  Action = 'action',
+  Informative = 'informative',
+  Link = 'link',
+}
+
+export enum TagSize {
+  Small = 'small',
+  Large = 'large',
 }
 
 export type TagProps = {
   /**
    * The label for the tag
    */
-  children: React.ReactNode;
+  children: string;
   /**
    * Additional class names to apply to the tag
    */
   className?: string;
   /**
-   * The aria-label for the delete button
-   * @deprecated Will be removed in the next major release.
+   * Url to go to after Link variant is clicked
    */
-  deleteButtonAriaLabel?: string;
+  href?: string;
   /**
-   * Prop will be passed to the delete button `<button>` element. It also hides the default label from screen readers to prevent confusion with labels when present.
-   * @deprecated Will be removed in the next major release.
+   * Link's _target -attribute
    */
-  deleteButtonProps?: React.ComponentPropsWithoutRef<'button'>;
+  target?: React.AnchorHTMLAttributes<HTMLAnchorElement>['target'];
+  /*
   /**
-   * Used to generate the first part of the id on the elements.
-   * Default value `hds-tag` will be removed in the next major release.
+   * Element placed on the starting side of the label
    */
-  id?: string;
+  iconStart?: React.ReactNode;
   /**
-   * Additional class names to apply to the tag's label element
-   * @deprecated Will be removed in the next major release.
+   * Element placed on the ending side of the label
    */
-  labelClassName?: string;
+  iconEnd?: React.ReactNode;
   /**
-   * Props that will be passed to the label `<span>` element.
-   * @deprecated Will be removed in the next major release.
+   * Should Tag span to multiple lines
+   * @default false
    */
-  labelProps?: React.ComponentPropsWithoutRef<'span'>;
+  multiline?: boolean;
   /**
    * Callback function fired when the tag is clicked. If set, the tag will be clickable.
    */
   onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => void;
   /**
-   * Callback function fired when the delete icon is clicked. If set, a delete button will be shown.
+   * Callback function fired when the tag is clicked. If set, a delete button will be shown.
    */
-  onDelete?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
-  /**
-   * Sets the role of the tag when it's clickable. Uses 'link' by default.
-   * @deprecated Usage will change in the next major release.
-   */
-  role?: 'link' | 'button';
+  onDelete?: (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => void;
```

Comment:
The old text is kind of misleading: "a delete button will be shown". Should be "delete icon"?

Oh, it is "delete icon" in docs.